ksqldb-execution/src/main/java/io/confluent/ksql/execution/plan/ExecutionStep.java

```diff
@@ -60,7 +60,13 @@
   @JsonIgnore
   List<ExecutionStep<?>> getSources();
 
-  S build(PlanBuilder planBuilder);
+  default S build(PlanBuilder planBuilder) {
```

Comment:
The current expectation is that this variant is only called by QueryExecutor, on output nodes, and not from any intermediate nodes. Is it worth enforcing this assumption by throwing an exception if the method is called a non-output-node execution step?

Reply:
+1. If we do not expect other callers, make it more strict is preferred.

Reply:
Turns out I was mistaken here. In the case of transient queries, there is no designated output node, so the QueryExecutor may call this variation of `build()` on essentially any node (whatever the last step in the execution step plan of a particular transient query happens to be).